Updated Events

orca277 opened the pull request:
# Description
- Modified files related to events by implementing addEvent, updateEvent, and deleteEvent.
- Deleted all documentation in event.dto.ts as I thought all the documentations as the variables seemed self-explanatory.
- Renamed variables in event.dto.ts as I thought it would be better to make the variable names be similar to the resources domain.
# Issue
-
# Testing
- Problem with event.controller.spec.ts which needs fixing

orca277 committed:
Updated Events

Modified files related to events by implementing addEvent, updateEvent, and deleteEvent.

orca277 committed:
Resolved Merge Conflict

Resolved Merge Conflict

orca277 committed:
Merge branch 'main' into feature/event

orca277 committed:
Update event.service.spec.ts

dev-taeho commented on src/domain/event/controller/event.controller.ts:
```diff
@@ -37,8 +51,35 @@ export class EventController {
+
+  /**
+   * @tag event
+   * @summary get events
```
Could you update swagger summary o accurately describe the endpoint?

dev-taeho commented on src/domain/event/dto/event.dto.ts:
```diff
@@ -1,85 +1,38 @@
-export interface GetEventsOutput {
-  /**
```
Could you clarify the reason for deleting all the documentation? Additionally, what was the motivation behind renaming the interface? If possible, could you write detail in pull request description?

orca277 replied:
Deleted all documentation as I thought all the documentations as I thought the variables were self-explanatory. Would you like me to put them all back?

For the part with renaming variables I wanted to make the variable names be unified with the resources domain which I was working.

dev-taeho replied:
Since we develop both the server and client, we understand the meaning and valid format of each field. However, when frontend and client developers are separate, it's important to provide clear documentation to ensure correct usage.
```
   * event start date time
   * example: "2024-12-27T00:00:00Z"
   * @type string
```
client won't know which is the valid format when they pass the start_date.

orca277 replied:
Has been readded

dev-taeho commented on src/domain/event/dto/event.dto.ts:
```diff
@@ -1,85 +1,38 @@
-   * example: 0
-   * @type number
-   */
+export interface EventsResponstDTO {
```
Can you correct typo?

dev-taeho commented on src/domain/event/repository/event.repository.spec.ts:
```diff
@@ -5,14 +5,16 @@ import {
 } from '@domain/event/repository/event.repository';
 
-describe('event repository', () => {
+describe('members repository', () => {
```
Can you correct it as `event`?

dev-taeho reviewed (changes requested):
Since `events` domain needs to handle images, could you update the service and controller to support image handling as well? You might need to integrate Cloudinary or an S3 bucket for image storage. Here are some relevant references:

- [NestJS File Upload Docs](https://docs.nestjs.com/techniques/file-upload)
- [Cloudinary](https://cloudinary.com/)

dev-taeho commented on src/domain/event/controller/event.controller.spec.ts:
```diff
@@ -58,7 +89,42 @@ describe('event controller', () => {
     const res = await request(app.getHttpServer()).get(`/v1/event/${data.id}`);
 
-    assertStatusCode(res, 200);
+    expect(res.statusCode).toEqual(200);
```
We use the [`assertStatusCode(res, statusCode)`](https://github.com/AKCSE-MB/AKCSE-Server/blob/main/jest.setup.ts#L10) test utility function for asserting status codes. Could you clarify the specific reason behind your decision to make changes?

orca277 replied:
The test fails were easier to read in the cli with expect()

dev-taeho replied:
Do you think we should standardize this approach across all tests, or would it be better to use `expect()` only in cases where readability is a concern? Alternatively, should we update all controller tests to use `expect()` for status code assertions?

What are your thoughts on the pros and cons of the `assertStatusCode` test utility function? Have you encountered any difficulties when checking status codes or function parameters? Do you have any suggestions for improving or modifying the function?

orca277 replied:
I don't believe it would be necessary for this approach. I've only changed it as this test was failing multiple times and personally for me the error descriptions were easier to read. If the failed test's log of assertStatusCode() is easier for the other majority of the developers then I believe it would be better to go with the majority.

dev-taeho replied:
If readability is a challenge, we could consider improving `assertStatusCode()` to provide better failure messages while keeping consistency across tests. Let me know your thoughts.

orca277 replied:
Yes I do think it's a good idea to keep consistency with testings. I'll look into ways with improving readability.

dev-taeho commented on src/domain/event/controller/event.controller.spec.ts:
```diff
@@ -24,6 +25,36 @@ describe('event controller', () => {
+        updatedAt: new Date(),
+      });
+
+    console.log(res);
```
Could you remove `console.log`?

dev-taeho replied:
```suggestion
```

could you remove `console.log`?

dev-taeho commented on src/domain/event/service/event.service.ts:
```diff
@@ -1,23 +1,62 @@
+  imageUrl?: string;
+  updatedAt: Date;
+}) {
+  return await saveEvent(param);
```
```suggestion
await saveEvent(param);
```
As discussed in this [PR](https://github.com/AKCSE-MB/AKCSE-Server/pull/39), we are not using the return value of `saveEvent` (`updateEvent`, and `deleteEvent`). Could you remove `return`?

orca277 committed:
Fixed minor issues

Removed console.log(res) in event.controller.spec.ts

Fixed typo in event.controller.ts

Fixed typo in event.dto.ts

Fixed description error in event.repository.spec.ts

Removed return in event.service.ts

orca277 committed:
Update event.dto.ts

Added descriptions for each interface variable in event.dto.ts

orca277 committed:
Update .env.sample

Added AWS credentials to github secrets and changed .env.sample accordingly.

orca277 committed:
Updated events domain to implement image upload

orca277 committed:
Update event.controller.ts

Updated event.controller.ts to properly utilize the image upload methods from event.service.ts

orca277 committed:
Updated npm dependencies

Installed aws-sdk and multer

orca277 committed:
Merge branch 'main' into feature/event

orca277 committed:
Update members.repository.spec.ts

Removed the expect() functions with individual checks as it was flagged with an error saying "res might be null"

orca277 committed:
Merge branch 'feature/event' of github.com:AKCSE-MB/AKCSE-Server into feature/event

orca277 committed:
Installed @sentry/nestjs

Installed @sentry/nestjs as running "npx jest 'event.controller.spec.ts'" was flagging an error that module @sentry/nestjs could not be found in line 13 of exception.filter.ts

orca277 committed:
Update members.repository.spec.ts

Removed the expects once again due to issues mentioned in commit cc95dce

dev-taeho commented on src/common/utils/s3.util.ts:
```diff
@@ -0,0 +1,26 @@
+import { S3 } from 'aws-sdk';
+
+const s3 = new S3({
```
In Nest.js, Configuration is usually handled by `ConfigurationService`. Please update to use `ConfigurationService` instead.

dev-taeho commented on package.json:
```diff
@@ -47,12 +47,16 @@
     "@sentry/profiling-node": "^8.51.0",
+    "@types/aws-sdk": "^2.7.4",
     "@types/bcrypt": "^5.0.0",
+    "aws-sdk": "^2.1692.0",
```
Please consider updating to AWS SDK v3
```
(node:2490) NOTE: The AWS SDK for JavaScript (v2) is in maintenance mode.
 SDK releases are limited to address critical bug fixes and security issues only.
``

orca277 committed:
Update .env.sample

Changed aws credential variables to placeholders

orca277 committed:
Added S3Service to app.module.ts

Modified s3.util.ts to utilize ConfigurationService.

Added S3Service to app.module.ts

Modified event.service.ts to use S3Service.

orca277 committed:
Updated aws-sdk

Removed aws-sdk and installed @aws-sdk/client-s3

dev-taeho commented on src/common/utils/s3.util.ts:
```diff
@@ -0,0 +1,45 @@
+import {
+  S3Client,
```
In our project,

**Domain**: Represents the areas we have divided within our service.
**Third-Party**: Contains adapters or implementations related to external services and packages.
Since implementing S3 services, which are third-party, I believe we should move this package under Third-Party

dev-taeho commented on src/domain/event/controller/event.controller.spec.ts:
```diff
@@ -24,6 +25,34 @@ describe('event controller', () => {
+        rsvpLink: '',
+        imageUrl: '',
+        createdAt: new Date(),
+        updatedAt: new Date(),
```
I believe `updatedAt` does not need to be passed as client input. instead, we can set it in the service layer.
